docs/src/pages/[platform]/components/icon/demo.tsx

```diff
@@ -1,96 +1,64 @@
 import * as React from 'react';
-import { Property } from 'csstype';
-
-import { Icon, TextField, View, useTheme } from '@aws-amplify/ui-react';
+import { Icon, IconProps } from '@aws-amplify/ui-react';
+import { IconPropControls } from './IconPropControls';
+import { useIconProps } from './useIconProps';
 import { Demo } from '@/components/Demo';
+import { demoState } from '@/utils/demoState';
 
-const propsToCode = ({ pathData, width, height, color, ariaLabel }) =>
-  `<Icon` +
-  (pathData ? `\n  pathData=${JSON.stringify(pathData)}` : '') +
-  `\n  viewBox={{` +
-  (width ? `\n    width: ${JSON.stringify(width)},` : '') +
-  (height ? `\n    height: ${JSON.stringify(height)},` : '') +
-  `\n  }}` +
-  (color ? `\n  color=${JSON.stringify(color)}` : '') +
-  (ariaLabel ? `\n  ariaLabel=${JSON.stringify(ariaLabel)}` : '') +
-  `\n/>`;
-
-const PropControls = (props) => {
-  const { tokens } = useTheme();
-  const {
-    width,
-    setWidth,
-    height,
-    setHeight,
-    ariaLabel,
-    setAriaLabel,
-    pathData,
-    setPathData,
-    color,
-    setColor,
-  } = props;
+const propsToCode = ({
+  pathData,
+  viewBox,
+  color,
+  width,
+  height,
+  ariaLabel,
+}: IconProps) => {
+  const hasViewBox = viewBox.width || viewBox.height;
   return (
-    <View padding={`${tokens.space.medium} 0`}>
-      <TextField
-        label="pathData"
-        value={pathData as string}
-        onChange={(event) => setPathData(event.target.value)}
-      />
-      <TextField
-        label="width"
-        value={width}
-        onChange={(event) => setWidth(event.target.value)}
-      />
-      <TextField
-        label="height"
-        value={height}
-        onChange={(event) => setHeight(event.target.value)}
-      />
-      <TextField
-        label="ariaLabel"
-        value={ariaLabel}
-        onChange={(event) => setAriaLabel(event.target.value)}
-      />
-      <TextField
-        label="color"
-        value={color}
-        onChange={(event) => setColor(event.target.value)}
-      />
-    </View>
+    `<Icon` +
+    (pathData ? `\n  pathData=${JSON.stringify(pathData)}` : '') +
```

Comment:
Just FYI, we have a `getPropString` utility function for handling this in `docs/src/pages/[platform]/components/utils/getPropString.ts`

Reply:
Nice! Updated to use that where I could in latest.